packages/mui-material/src/Modal/useModal.ts

```diff
@@ -83,7 +85,13 @@ function useModal(parameters: UseModalParameters): UseModalReturnValue {
   };
 
   const handleOpen = useEventCallback(() => {
-    const resolvedContainer = getContainer(container) || getDoc().body;
+    /**
+     * Resolving this could be simplified (mountNodeRef should take priority when it's set)
+     * but this will also work because the logic matches {@link Portal}.
+     */
+    const resolvedContainer = disablePortal
+      ? ((mountNodeRef.current ?? modalRef.current)?.parentElement ?? getDoc().body)
+      : getContainer(container) || getDoc().body;
```

Comment:
If providing the `container` prop was required when using `disablePortal`, this could be simplified to 

```suggestion
const resolvedContainer = getContainer(container) || getDoc().body;
```

Right?

Reply:
But it's not required and it shouldn't be, because if you don't use Portal how could you mount it to a different container?

That is also the reason `disablePortal` takes priority over `container` here.